### ISSUE TITLE
Accessibility Audit: Improper announcement of dialog when user launches filter modal

### ISSUE THREAD

**Issue by jnowell** [CityOfDetroit/bloom#952]
**ENVIRONMENT** 
macOS 11.6 | Chrome Version 95.0.4638.69 (Official Build) (x86_64) | Voiceover
Chrome OS Version 93.0.4577.51 (Official Build) beta (64-bit) | Chromevox
Windows 10 Enterprise | Chrome Version 96.0.4664.45 (Official Build) (64-bit) | JAWS
Windows 10 Enterprise | Chrome Version 96.0.4664.45 (Official Build) (64-bit) | NVDA
Windows 10 Enterprise | Firefox Version 92.0.2 (64-bit) | JAWS

**ISSUE**

Turn on the Screen Reader, and go to the Listings page. Open the 'Filter Results' modal. Observe the behavior and announcement when filter results dialog opens up. As you can see from the attached video, the Screen reader doesn't announce the dialog in a proper manner. Voiceover announces 'and 2 more items' dialog. Chromevox & JAWS announces all content of the dialog. NVDA doesn't announce the label of the dialog, it announces the control on which the focus is placed - unit availability.

Screen reader should announce the dialog label in a proper manner. Ideally the dialog can be announced with the focus placed at the top of the dialog or dialog can be announced with focus placed at the first control of the dialog.

https://user-images.githubusercontent.com/394908/148572726-1867213a-29f8-4188-acab-d2296a2655d5.mov
